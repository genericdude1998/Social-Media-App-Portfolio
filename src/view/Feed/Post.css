@import '../globalStyles//cssVars.css';

*{
    box-sizing: border-box;
    padding: 0px; 
}

.basePost{
    background-color: royalblue;
    height: 30vh;
    width: 100vw;
}

.basePostContent{
    margin: 10px;
    color: white;
    line-height: 50px;
}

.basePostDate{
    color:white;
    text-align: right;
    margin: 30px;
}

.basePostLink{
    color: darkblue;
    font-weight: bold;
    text-decoration: none;
}

.dark{
    background-color: black;
    color: white;
}

