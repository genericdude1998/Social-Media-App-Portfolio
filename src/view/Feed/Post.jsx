import React from 'react';
import styles from './Post.css'

const Post = ({name, content, date}) => {
    return (
        <div className={styles.basePost}>
            <h2 className={styles.basePostTitle}>{name}</h2>
            <p className={styles.basePostContent}>{content}</p>
            <p className={styles.basePostDate}>{date}</p>
        </div>
    );
}

export default Post;
