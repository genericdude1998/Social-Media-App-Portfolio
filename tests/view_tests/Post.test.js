import React from 'react'
import { shallow } from 'enzyme';
import Post from '../../src/view/Feed/Post';
import {mockContent, mockDate, mockName} from '../mockValues';


let wrapper = shallow(
    <Post
        name={mockName}
        content={mockContent}
        date={mockDate}
    />
);

describe('Post', () => {
    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});