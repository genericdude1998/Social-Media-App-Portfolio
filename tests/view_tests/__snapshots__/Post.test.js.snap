// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should match snapshot 1`] = `
<div
  className="basePost"
>
  <h2
    className="basePostTitle"
  >
    mock_name
  </h2>
  <p
    className="basePostContent"
  >
    mock_content
  </p>
  <p
    className="basePostDate"
  >
    mock_date
  </p>
</div>
`;
